### PR TITLE
Interface

### PR DESCRIPTION
**Pull request by nicolaslesquoy**
Nouvelle interface et "amélioration du code".
- Séparation des fonctions en fichiers et en classe
- Interface construire sur `rich`.
- Reprise des fonctionnalités initiales et ajout de fonctions de sauvegarde et de modifications des paramètres

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# LaTeX Project-Manager
+*A project-manager tool to speed up creation process for note-taking*
+---
+## **Description**
+LPM (LaTeX Project Manager) is a simple project manager to generate a new LaTeX project from a template.
+
+## Requirements
```
Comment (amtoine) [inline README.md:7]: always great to have a "requirements" section :ok_hand: 

for `python`, i recommend writing a `requirements.txt` file with the list of dependencies, here
```
# requirements.txt
art
rich
```
and then install it with `pip install -r requirements.txt` :+1:

Comment (amtoine) [reply README.md:7]: and regarding the next lines, you can have a look to `python`'s virtual environments, namely `virtualenv` and `pipenv`, which builds upon `virtualenv`, both very handy :ok_hand:

File: README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+- `art` library (`python3 -m pip install art`)
+
+Tested on *Ubuntu 22.04 LTS*. 
+  
+## Installation
+```
+git clone https://github.com/iScsc/LaTeX-Project-Manager.git
+cd ./LaTeX-Project-Manager/settings
+```
+- Change the "PathToSource" with the *absolute* path to the `source_TeX` folder in the `/settings/config.json` file.
```
Comment (amtoine) [inline README.md:18]: that is very tideous to do... :thinking: 
maybe you could:
- write `$HOME/path/to/source` inside the file and parse the `$HOME` inside the main code with the `expanduser` function of built-in `os` or `pathlib` libraries
- make your project a real `python` package that you could install with `pip install -e .`. then, the path to the source would not matter :+1:

File: README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+```
+cd ../src
+```
+- Change the path with the *absolute* path to the `config.json` file. 
+```python
+SettingsLaunch = Settings(r"/home/nlesquoy/ghq/LaTeX-Project-Manager/settings/config.json")
```
Comment (amtoine) [inline README.md:24]: this is a remark for the whole code base: `python` has a guideline called `PEP8`, with both code and style rules.
these rules does not HAVE to be followed, but the thing that everyone, me included, use them and a code which violates these rules is super hard to read and appreciate, even if great...

one of these rules is to prefer _snake\_case_ over _CamlCase_ for everything, except class names, including variables, functions and files :+1: 

i recommend updating the code base in that direction

and as a general rule of thumb, you can use `flake8` and `black` to lint and format your `python` source files respectively :+1:

File: src/commit.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# `commit.md` - Rules for Git commit messages
```
Comment (amtoine) [inline src/commit.md:1]: such a file might not want to go in `src` but rather in `res`?
not a big deal, that's just not really a source file :wink:

File: src/convention.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# `convention.md` - How should your file be named ?
```
Comment (amtoine) [inline src/convention.md:1]: this file looks like a duplicate of `src/commit.md`

File: src/app.cfg [hunk 6]
```diff
@@ -0,0 +1,2 @@
+/home/nlesquoy/ghq/github.com/LaTeX-Project-Manager/settings/config.json
```
Comment (amtoine) [inline src/app.cfg:1]: same remark as above regarding the hardcoded paths :thinking: 
you could parse `$HOME` inside the code :+1:

**Review by amtoine — CHANGES_REQUESTED** [review]
i haven't looked at the source files for now, to let you change the things you want to change with the current comments i've made :ok_hand: 

also, something i did not mention in the review comments, you mix french and english in the code, the other files, the _GitHub_ side and the commits, which is quite confusing... i think choosing a side (english :pleading_face:) would make things clearer :yum: 

i think it will be easier to review the code after these changes :yum:

File: README.md (hunk 3 above)
Comment (nicolaslesquoy) [reply README.md:24]: Indeed I will use `flake8` to clean the code. I will also rewrite the docstring for another documentation tool (the current tool used here is `sphinx`).

File: src/commit.md (hunk 4 above)
Comment (nicolaslesquoy) [reply src/commit.md:1]: I will move it to a `res` folder.

File: src/convention.md (hunk 5 above)
Comment (nicolaslesquoy) [reply src/convention.md:1]: That is not a duplicate. I use two conventions, although very similar. There is one convention for the naming of the file, and another for the commit used to save said files.

File: README.md (hunk 3 above)
Comment (amtoine) [reply README.md:24]: `sphinx` is fine, but change the framework if you fell more comfortable with another one :ok_hand:

Comment (amtoine) [reply README.md:24]: i also quite like the [Google Style Python Docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) if you do not already know it :wink:

File: src/commit.md (hunk 4 above)
Comment (amtoine) [reply src/commit.md:1]: looking forward to the change :yum:

File: src/convention.md (hunk 5 above)
Comment (amtoine) [reply src/convention.md:1]: mm, these are actually the same file, only the header changes, but the list is the exact same one :thinking: 
maybe i misunderstand something here though

**Pull request merged**